Source/Common/CommonBase.cpp

```diff
@@ -98,15 +98,26 @@ void CommonBase::MPISpinner()
 
 int MSGBUFSIZ = 1024;                  // Individual incoming message buffer
 char * MSGBUF = new char[MSGBUFSIZ];   // Pull it off the heap
+
+const int MAX_PROBE_IDLE_BACKOFF_US=100000; // Maximum is 100,000us, i.e. 10hz
+int probe_idle_backoff_us=0;
 for (;;) {
 // See if there are any MPI packets coming down the pipe
   MPI_Status status;                   // Note the multi-threaded MPI probe
   int flag;
   MPI_Iprobe(MPI_ANY_SOURCE,MPI_ANY_TAG,MPI_COMM_WORLD,&flag,&status);
   if (flag==0) {                       // Nothing there....
-    OnIdle();                          // Guess
+    if(HaveIdleWork()){
+      OnIdle();                          // Guess

```

Comment:
Just having a think about this - shouldn't the sleep period be reset if there is idle work and `OnIdle()` is called?

Currently, the period is incremented any time there is no idle work, but it is not reset if there is.

Reply:
My mental model was that the OnIdle was mostly draining work that arrived in messages, so it was something like:
- Message arrives, causing work to queue up, resetting delay to zero
- HaveIdleWork returns true until all the work is finished, with delay staying at zero
- HaveIdleWork returns false (and presumably keeps returning false), so delay slowly increases

But if idle work can arrive from elsewhere (i.e. not from a message), then I guess that you'd
want any return from HaveIdleWork to reset the counter too.